gcc/rust/typecheck/rust-tyty-call.cc

```diff
@@ -23,6 +23,39 @@
 namespace Rust {
 namespace TyTy {
 
+void
+emit_unexpected_argument_error (Location loc,
+				unsigned long unexpected_arg_count,
+				unsigned long expected_arg_count)
+{
+  // https://doc.rust-lang.org/error_codes/E0061.html
+  // rustc treats 1 as singular and others as plural
+  std::string err_msg = "this function takes %lu ";
+
+  if (expected_arg_count == 1)
+    {
+      err_msg += "argument";
+    }
+  else
+    {
+      err_msg += "arguments";
+    }
+
+  if (unexpected_arg_count == 1)
+    {
+      err_msg += " but %lu argument was supplied";
+    }
+  else
+    {
+      err_msg += " but %lu arguments were supplied";
+    }
+
+  rust_error_at (loc, ErrorCode ("E0061"), err_msg.c_str (), expected_arg_count,
+		 unexpected_arg_count);
+
+  return;
```

Comment:
you can remove the return statement and I will merge

Reply:
Done :)